src/bootstrap/install.rs

```diff
@@ -210,10 +210,14 @@ install!((self, builder, _config),
         }
     };
     LlvmTools, alias = "llvm-tools", Self::should_build(_config), only_hosts: true, {
-        let tarball = builder
-            .ensure(dist::LlvmTools { target: self.target })
-            .expect("missing llvm-tools");
-        install_sh(builder, "llvm-tools", self.compiler.stage, Some(self.target), &tarball);
+        if let Some(tarball) = builder.ensure(dist::LlvmTools { target: self.target }) {
+            install_sh(builder, "llvm-tools", self.compiler.stage, Some(self.target), &tarball);
+        } else {
+            // LlvmTools can be externally provided
+            builder.info(
+                &format!("skipping Install LlvmTools stage{} ({})", self.compiler.stage, self.target),
+            );
+        }
```

Comment:
We should check if llvm is provided or not. If llvm isn't provided externally and there is a problem with tarballs, we wan't to panic the process instead of skipping it. With this diff any problem with tarball will skip the process regardless.

Reply:
I agree, but I don't know how to do what you suggest.
All other installs (Miri, etc..) either panic or skip if the tarball is not defined.

Reply:
You can implement a function in `Config` that checks if any of the `configure_args` items starts with `--llvm-root`, and returns bool value depend on that.

Reply:
Done. Had to move some code around.